pkgs/applications/audio/rhythmbox/default.nix

```diff
@@ -33,6 +34,8 @@ in stdenv.mkDerivation rec {
     wrapGAppsHook
   ];
 
+  propagatedBuildInputs = [ libdmapsharing ];
```

Comment:
Why are you propagating it?

Reply:
Maybe I don't understand what propagating means. My intent is that environments which include rhythmbox should also include libdmapsharing.

Reply:
`propagatedBuildInputs` do not do that. They only make it so that if you add `rhythmbox` to `buildInputs` of some other package, it would be as if you added `libdmapsharing` to its `buildInputs` as well.

Here you simply want a C program to be linked against a C library (I assume) so `buildInputs` should be sufficient. The compiler/linker will ensure that the library path is stored in the `DT_RUNPATH` entry of the resulting ELF file – no need to deal with any environments.